signbank/dictionary/csv_import.py

```diff
@@ -231,10 +231,23 @@ def import_nzsl_share_gloss_csv(request):
         for row in glossreader:
             if glossreader.line_num == 1:
                 continue
-            if row["id"] not in existing_nzsl_share_ids:
-                new_glosses.append(row)
+            if row["id"] in existing_nzsl_share_ids:
+
+                # nzsl_share_id is not a reliable index, due to manual intervention
```

Comment:
note: I think we should follow up about that, if we think its feasible for it to be a reliable index

the impression I got from the clients comment about this was originally they were having to manually input data before we implemented this importing process, and that's how we ended up with some records with multiple or duplicate `nzsl_share_id` values.

if that's the case, we should check if going forward our desire is to have `nzsl_share_id` unique in which case we should consider cleaning up the old records that are currently not unique so that going forward we can properly rely on this

Reply:
Client has confirmed that the `nzsl_share_id` should be unique going forward, and that they're keen to get the existing duplicates cleaned up:

> Yes, I think nzsl_share_id should be unique going forward. I can clean up the duplicates and multiples.

ideally it would be good to enforce that at a database level but even if that's easily done it's probably best tackled in a dedicated PR so for now let's just ensure we graceful skip/fail for this PR

Reply:
>  let's just ensure we graceful skip/fail for this PR
So - the way I have it working? Or something else?

Reply:
Yeah this is more of a "meta comment" rather than requiring a code change within this specific PR

Reply:
Closing this thread prior to pushing.